### ISSUE TITLE
Xamarin Null Reference Error - 0.8.6 - beta 2

### ISSUE THREAD

**Issue by softsan** [ably/ably-dotnet#126]
We have our app crash with  following exception  in  PROD env from Xamarin.Forms app.  We are using ably 0.8.6 - beta 2

````
Android: 6.0.1
Android Build: MCG24.251-5-5
Manufacturer: motorola
Model: XT1254
Thread: Thread-24982-24982
Start Date: Tue Aug 22 18:05:20 CDT 2017
Date: Tue Aug 22 18:05:28 CDT 2017
Format: Xamarin

Xamarin caused by: android.runtime.JavaProxyThrowable: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> IO.Ably.AblyException: Reason: Unexpected error :Object reference not set to an instance of an object; Code: 50000 ---> System.NullReferenceException: Object reference not set to an instance of an object
  at IO.Ably.Realtime.Presence+PresenceMap.EndSync () [0x00090] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Realtime.Presence.OnPresence (IO.Ably.PresenceMessage[] messages, System.String syncChannelSerial) [0x00097] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Realtime.ChannelMessageProcessor.MessageReceived (IO.Ably.Types.ProtocolMessage protocolMessage) [0x00182] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Transport.ConnectionManager+<ProcessTransportMessage>d__73.MoveNext () [0x000b4] in <8de47a40057d4140b308beeae794bcf0>:0 
   --- End of inner exception stack trace ---
  at IO.Ably.Transport.ConnectionManager+<ProcessTransportMessage>d__73.MoveNext () [0x000e0] in <8de47a40057d4140b308beeae794bcf0>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <156481a94d804d75be4f89f45f73eeca>:0 
  at IO.Ably.Transport.ConnectionManager+<OnTransportMessageReceived>d__71.MoveNext () [0x000cd] in <8de47a40057d4140b308beeae794bcf0>:0 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) IO.Ably.AblyException: Reason: Unexpected error :Object reference not set to an instance of an object; Code: 50000 ---> System.NullReferenceException: Object reference not set to an instance of an object
  at IO.Ably.Realtime.Presence+PresenceMap.EndSync () [0x00090] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Realtime.Presence.OnPresence (IO.Ably.PresenceMessage[] messages, System.String syncChannelSerial) [0x00097] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Realtime.ChannelMessageProcessor.MessageReceived (IO.Ably.Types.ProtocolMessage protocolMessage) [0x00182] in <8de47a40057d4140b308beeae794bcf0>:0 
  at IO.Ably.Transport.ConnectionManager+<ProcessTransportMessage>d__73.MoveNext () [0x000b4] in <8de47a40057d4140b308beeae794bcf0>:0 
   --- End of inner exception stack trace ---
  at IO.Ably.Transport.ConnectionManager+<ProcessTransportMessage>d__73.MoveNext () [0x000e0] in <8de47a40057d4140b308beeae794bcf0>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <156481a94d804d75be4f89f45f73eeca>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <156481a94d804d75be4f89f45f73eeca>:0 
  at IO.Ably.Transport.ConnectionManager+<OnTransportMessageReceived>d__71.MoveNext () [0x000cd] in <8de47a40057d4140b308beeae794bcf0>:0 <---
```

**Comment by mattheworiordan** [comment]
Note from @marto83 on this issue:

> There is a missing null check for residualMembers here https://github.com/ably/ably-dotnet/blob/1baa3bd069c11d4180c7e2055a29209c7d3049a6/src/IO.Ably.Shared/Realtime/Presence.cs#L369

To be addressed when this issue is resolved.

**Comment by marto83** [comment]
I've released 0.8.5. Please use that version instead the bug is fixed there.

**Comment by softsan** [comment]
We have this error on 0.8.5.
```
The method or operation is not implemented.
System.Net.Sockets.MultipleConnectAsync.AsyncFail(Exception e)<081f9cb1e65248a1910ff4c949501a6a>:0
System.Net.Sockets.MultipleConnectAsync.Fail(bool sync, Exception e)<081f9cb1e65248a1910ff4c949501a6a>:0
System.Net.Sockets.MultipleConnectAsync.DoDnsCallback(IAsyncResult result, bool sync)<081f9cb1e65248a1910ff4c949501a6a>:0
System.Net.Sockets.MultipleConnectAsync.DnsCallback(IAsyncResult result)<081f9cb1e65248a1910ff4c949501a6a>:0
System.Runtime.Remoting.Messaging.AsyncResult.Invoke(AsyncResult)(wrapper managed-to-native)
System.Runtime.Remoting.Messaging.AsyncResult.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()<b9aa7fa9d054435b96a2da7421b28c4d>:0
System.Threading.ThreadPoolWorkQueue.Dispatch()<b9aa7fa9d054435b96a2da7421b28c4d>:0
System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()<b9aa7fa9d054435b96a2da7421b28c4d>:0
```

**Comment by marto83** [comment]
@softsan I've created a separate issue #129 and added a comment. 

Either use 0.8.5-rc3 which doesn't update the Websocket4Net reference or you need to set this permission `android.permission.INTERNET` in your manifest file.
